Gems/Atom/Utils/Code/Include/Atom/Utils/ImGuiGpuProfiler.inl

```diff
@@ -1025,6 +1060,221 @@ namespace AZ
             }
         }
 
+        // --- ImGuiGpuMemoryView ---
+
+        inline void ImGuiGpuMemoryView::SortTable(ImGuiTableSortSpecs* sortSpecs)
+        {
+            const bool ascending = sortSpecs->Specs->SortDirection == ImGuiSortDirection_Ascending;
+            const ImS16 columnToSort = sortSpecs->Specs->ColumnIndex;
+
+            // Sort by the appropriate column in the table
+            switch (columnToSort)
+            {
+            case (0): // Sorting by parent pool name
+                AZStd::sort(m_tableRows.begin(), m_tableRows.end(),
+                    [ascending](const TableRow& lhs, const TableRow& rhs)
+                    {
+                        const auto lhsParentPool = lhs.m_parentPoolName.GetStringView();
+                        const auto rhsParentPool = rhs.m_parentPoolName.GetStringView();
+                        return ascending ? lhsParentPool < rhsParentPool : lhsParentPool > rhsParentPool;
+                    });
+                break;
+            case (1): // Sort by buffer/image name
+                AZStd::sort(m_tableRows.begin(), m_tableRows.end(),
+                    [ascending](const TableRow& lhs, const TableRow& rhs){
+                        const auto lhsName = lhs.m_bufImgName.GetStringView();
+                        const auto rhsName = rhs.m_bufImgName.GetStringView();
+                        return ascending ? lhsName < rhsName : lhsName > rhsName;
+                    });
+                break;
+            case (2): // Sort by memory usage
+                AZStd::sort(m_tableRows.begin(), m_tableRows.end(),
+                    [ascending](const TableRow& lhs, const TableRow& rhs){
+                        const float lhsSize = lhs.m_sizeInBytes;
+                        const float rhsSize = rhs.m_sizeInBytes;
+                        return ascending ? lhsSize < rhsSize : lhsSize > rhsSize;
+                    });
+                break;
+            }
+            sortSpecs->SpecsDirty = false;
+        }
+
+        inline void ImGuiGpuMemoryView::DrawTable()
+        {
+            if (ImGui::BeginTable("Table", 4, ImGuiTableFlags_Borders | ImGuiTableFlags_Sortable | ImGuiTableFlags_Resizable))
+            {
+                ImGui::TableSetupColumn("Parent pool");
+                ImGui::TableSetupColumn("Name");
+                ImGui::TableSetupColumn("Size (MB)", 0, 100.0f);
+                ImGui::TableSetupColumn("BindFlags", ImGuiTableColumnFlags_NoSort);
+                ImGui::TableHeadersRow();
+                ImGui::TableNextColumn();
+
+                ImGuiTableSortSpecs* sortSpecs = ImGui::TableGetSortSpecs();
+                if (sortSpecs && sortSpecs->SpecsDirty){
```

Comment:
Nit Coding style for consistency: Bring the { down to a new line. Do this everywhere else as well.